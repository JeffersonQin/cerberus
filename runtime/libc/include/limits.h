#ifndef	_LIMITS_H_
#define	_LIMITS_H_

#define CHAR_BIT   __cerbvar_CHAR_BIT
#define SCHAR_MIN  __cerbvar_SCHAR_MIN
#define SCHAR_MAX  __cerbvar_SCHAR_MAX
#define UCHAR_MAX  __cerbvar_UCHAR_MAX
#define CHAR_MIN   __cerbvar_CHAR_MIN
#define CHAR_MAX   __cerbvar_CHAR_MAX
#define MB_LEN_MAX __cerbvar_MB_LEN_MAX
#define SHRT_MIN   __cerbvar_SHRT_MIN
#define SHRT_MAX   __cerbvar_SHRT_MAX
#define USHRT_MAX  __cerbvar_USHRT_MAX
#define INT_MIN    __cerbvar_INT_MIN
#define INT_MAX    __cerbvar_INT_MAX
#define UINT_MAX   __cerbvar_UINT_MAX
#define LONG_MIN   __cerbvar_LONG_MIN
#define LONG_MAX   __cerbvar_LONG_MAX
#define ULONG_MAX  __cerbvar_ULONG_MAX
#define LLONG_MIN  __cerbvar_LLONG_MIN
#define LLONG_MAX  __cerbvar_LLONG_MAX
#define ULLONG_MAX __cerbvar_ULLONG_MAX


// POSIX
#define IOV_MAX    1024
#define NAME_MAX   1024

#endif
