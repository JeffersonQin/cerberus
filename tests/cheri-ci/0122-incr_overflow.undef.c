#include <limits.h>
int main(void)
{
  int x = INT_MAX;
  signed char c = SCHAR_MAX;
  c++; // defined because of the promotion
  x++;
}
