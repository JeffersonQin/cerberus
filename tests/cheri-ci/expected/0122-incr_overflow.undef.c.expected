Undefined {ub: "UB036_exceptional_condition", stderr: "", loc: "<7:3--7:6>"}
